python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -71,53 +74,17 @@ def __init__(self, provider_config, cluster_name):
         self.tag_cache = {}  # Tags that we believe to actually be on EC2.
         self.tag_cache_pending = {}  # Tags that we will soon upload.
         self.tag_cache_lock = threading.Lock()
-        self.tag_cache_update_event = threading.Event()
-        self.tag_cache_kill_event = threading.Event()
-        self.tag_update_thread = threading.Thread(
-            target=self._node_tag_update_loop)
-        self.tag_update_thread.start()
+        self.tag_update_thread = threading.Thread()
+        self.tag_update_thread.start()  # Put this thread in finished state.
```

Comment:
would this thread be leaked?

Reply:
I don't think so? Since the start method isn't passed a target, the thread goes immediately into a stopped state. I think the thread should then be garbage-collected after `self.tag_update_thread` is reassigned in `set_node_tags` or when the `AWSNodeProvider` is destroyed. Why might the thread be leaked?